File: src/game/entity/enemies/npcs/Helicopter.java
package game.entity.enemies.npcs;

import game.utils.Paths;

public class Helicopter extends FlyingEnemy {
    public Helicopter() {
        super();
    }

    @Override
    protected String getBasePath() {
        return Paths.getEnemiesFolder() + "/heli";
    }

    public String[] getCharacterOrientedImages() {
        return new String[]{
                String.format("%s/heli_%s_0.gif", getBasePath(), imageDirection.toString().toLowerCase()),
                String.format("%s/heli_%s_1.gif", getBasePath(), imageDirection.toString().toLowerCase()),
                String.format("%s/heli_%s_2.gif", getBasePath(), imageDirection.toString().toLowerCase()),
        };
    }

}

File: src/game/entity/enemies/npcs/Helicopter.kt
package game.entity.enemies.npcs

import game.entity.EntityTypes
import game.utils.Paths.enemiesFolder

class Helicopter() : FlyingEnemy() {
    constructor(id: Long) : this() {
        this.id = id
    }

    override fun getEntitiesAssetsPath(): String = "$enemiesFolder/heli"

    override fun getCharacterOrientedImages(): Array<String> = Array(3) { index ->
        "$entitiesAssetsPath/heli_${imageDirection.toString().lowercase()}_$index.gif"
    }

    override fun getType(): EntityTypes = EntityTypes.Helicopter
}